From 8fdfdcfd47860fa3b1ca6a35b292e9dcedc4c16d Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Wed, 18 Sep 2019 22:33:42 +0200
Subject: [PATCH] [addons] fix addon version.h, do not give nullptr as string

**Fix with very high priority**
---------------------------------

The change on https://github.com/xbmc/xbmc/pull/16591/files#diff-76f1176ce521a817a7f4c9d779760fe1R281
destroy all Kodi's before, if a binary repo addon was created after this request, it will crash
immediately with nullptr to std::string.

Therefore, no older Kodi would work anymore!
---
 xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
index 5cf34c6f1586..d95a74729789 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
@@ -278,7 +278,7 @@ inline const char* GetTypeVersion(int type)
       return ADDON_INSTANCE_VERSION_VIDEOCODEC;
 #endif
   }
-  return nullptr;
+  return "0.0.0";
 }
 
 ///
@@ -331,7 +331,7 @@ inline const char* GetTypeMinVersion(int type)
     case ADDON_INSTANCE_VIDEOCODEC:
       return ADDON_INSTANCE_VERSION_VIDEOCODEC_MIN;
   }
-  return nullptr;
+  return "0.0.0";
 }
 
 ///
